controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -312,7 +312,92 @@ void wfa_ca::handle_wfa_ca_message(
+        *               |                               | SSID, authentication type (WPA2-PSK),
+        *               |                               | encryption type(AES-CCMP), network key, 
+        *               |                               | bit 6 of Multi-AP IE's extension
+        *               |                               | attribute, bit 5 of Multi-AP IE's,
+        *               |                               | bit 5 of Multi-AP IE's extension
+        *               |                               | attribute. To clear the BSS info stored
+        *               |                               | a specific operating class, only the
+        *               |                               | device ALID and operating class are 
+        *               |                               | retained.
+        *               |                               |
+        * "bss_info2"   | string                        |  
+        * "bss_infoN"   | string                        |  
+        * "name"        | string (ex: "dev1")           | Device name
+        * "program"     | string ("map)                 | Program name
+        * 
+        * Return values:
+        *   None.
+        * 
+        * 
+        * Example:
+        *   UCC: dev_set_config,name,agt1,program,map,backhaul,0x00904C2A11B7
+        *   CA:status,RUNNING
+        *   CA:status,COMPLETE
+        */
+
+        // NOTE: Note sure this parameters are actually needed. There is a controversial
+        // between TestPlan and CAPI specification regarding if these params are required.
+        static std::unordered_map<std::string, std::string> params{
+            // {"name", std::string()},
+            // {"program", std::string()},
+        };
+
+        // NOTE: Uncertainty if param names: "backhaul", "bss_info{1..N}" are mandatory
+
+        if (!parse_params(params, err_string)) {
+            LOG(ERROR) << err_string;
+            reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, err_string);
+            break;
+        }
+
+        // Input check
+        if (params.find("program") != params.end()) {
+            if (params["program"] != "map") {
+                err_string = "invalid param value '" + params["map"] +
+                             "' for param name 'program', accepted value can be only 'map'";
+                LOG(ERROR) << err_string;
+                reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, err_string);
+                break;
+            }
+        }
+
+        // TODO: Find out what to do with value of params: "backhaul", bss_info{1..N}, "name".
+
+        // Send back first reply
+        if (!reply_running(sd, cmdu_tx)) {
+            LOG(ERROR) << "failed to send reply";
+            break;
+        }
+
+        // TODO: Configure device "name". Currently not needed.
+        // NOTE: This action is an asynchronous procedure and will be required to be executed
+        //       from task infrastructure.
```

Comment:
It is an asynchronous process since you need to config an Agent which requires to send a message to the agent and get back a response, and when all of this is done you return "complete".

And yes, the "running" is needed after validity check of the command, and on commands that it may take a lot of time to return a response - like configuring an agent.
For commands like "dev_get_param" you can return an immediate complete status since all the data is locally on the controller and it doesn't need to ask the agent to send it to him.

Reply:
> It is an asynchronous process since you need to config an Agent which requires to send a message to the agent and get back a response, and when all of this is done you return "complete".

 OK, then I misunderstood the meaning of this CAPI command. I thought it was meant to set the parameters locally in the controller, not to go and configure the agents. I thought the configuration would be triggered by a send_1905 autoconfig_renew or by triggering an autoconfig on the agent itself.